OSDOCS-1196 Custom endpoint configuration for route53 (#26270)
openshift/openshift-docs: modules/nw-endpoint-route53.adoc

// Module included in the following assemblies:
//
// * installing/installing_aws/installing-aws-government-region.adoc

[id="nw-endpoint-route53_{context}"]
= Endpoint configuration for route53

> The `route53` endpoint URL must use regions, `us-east-1`, `us-west-1`, or no region because Route 53 is not a regionalized service. The `us-east-1` region is required for the `tagging` endpoint URL to get the hosted zone, `route 53`, of the Elastic Load Balancing (ELB).

Author: I think we can remove the "because Route 53 is not a regionalized service" language, but I'm not sure yet.

[IMPORTANT]
====

> Private, disconnected installs are not supported for AWS GovCloud when installing in the `us-gov-east-1` region.

Reviewer: I'm not aware of this limitation. Users should be able to install in either GovCloud region without the need to specify any custom endpoints.

Author: Thank you for reviewing, @danehans! I pulled the note from the known issue that came up on Friday of last week: https://github.com/openshift/openshift-docs/pull/26300, but am happy to remove it. 

Do you think this content is important for the user when installing on GovCloud? The intention for this module is to include it in the AWS GovCloud installation guide. 

Currently the into paragraph is: 
The `route53` endpoint URL must use regions `us-west-1. The `us-west-1` region is required for the `tagging` endpoint URL to get the hosted zone, `route 53`, of the Elastic Load Balancing (ELB).

Would it be better to include your breakdown or replace the introduction all together: 
"If you install in either GovCloud region, Ingress Operator uses `us-gov-west-1` region for Route53 and tagging API clients.

Ingress Operator uses `https://tagging.us-gov-west-1.amazonaws.com` as the tagging API endpoint if a tagging custom endpoint is configured that includes the string 'us-gov-east-1'."

Reviewer: > Thank you for reviewing, @danehans! I pulled the note from the known issue that came up on Friday of last week: #26300, but am happy to remove it.

@bmcelvee thanks for the heads-up. I was unaware of that bz, so please keep the note.

> Do you think this content is important for the user when installing on GovCloud? The intention for this module is to include it in the AWS GovCloud installation guide.

I don't think so, the caveats stated are specific to "custom endpoints" and according to @abhinavdahiya custom endpoints are not required for GovCloud.

> Would it be better to include your breakdown or replace the introduction all together:
"If you install in either GovCloud region, the Ingress Operator uses us-gov-west-1 region for Route53 and tagging API clients.

Yes, I like this replacement.

====

.Example `route53` configuration
[source,yaml]
----
platform:
  aws:
    region: us-gov-west-1
    serviceEndpoints:
    - name: ec2
       url: https://ec2.us-gov-west-1.amazonaws.com
    - name: elasticloadbalancing
       url: https://elasticloadbalancing.us-gov-west-1.amazonaws.com

>     - name: route53 # Route 53 is not regionalized, so us-east-1 or no region.

Reviewer: I suggest removing `# Route 53 is not regionalized, so us-east-1 or no region.`.

>        url: https://s3.us-east-1.amazonaws.com

Reviewer: substitute s3 for route53.

Author: Subbed

Reviewer 2: seems above was not updated? should be
```
        - name: route53
          url: https://route53.us-gov.amazonaws.com
```
please check https://bugzilla.redhat.com/show_bug.cgi?id=1866299 and it is `us-gov` for route53, no suffix `-west-1`.

>     - name: tagging # us-east-1 is required to get the hosted zone id of the LB for the ingress domain.

Reviewer: I suggest removing `# us-east-1 is required to get the hosted zone id of the LB for the ingress domain.`

Author: Removing

// Module included in the following assemblies:
//
// * installing/installing_aws/installing-aws-government-region.adoc

[id="nw-endpoint-route53_{context}"]
= Endpoint configuration for route53

> If you install in either GovCloud region, the Ingress Operator uses `us-gov-west-1` region for Route53 and tagging API clients.

Reviewer 2: Checked https://bugzilla.redhat.com/show_bug.cgi?id=1866299 (Comment 40 and 58) , for Route53 it always uses `https://route53.us-gov.amazonaws.com`, but tagging API follows the region name and can use `https://tagging.us-gov-east-1.amazonaws.com` or `https://tagging.us-gov-west-1.amazonaws.com`.

Reviewer 2: To avoid confusion, we should tell user the difference between the endpoints user specified in yaml file and the ones Ingress operator uses in implementation. 
And user should specify Route53 as `https://route53.us-gov.amazonaws.com` for both GovCloud region, see https://docs.aws.amazon.com/govcloud-us/latest/UserGuide/using-govcloud-endpoints.html, only US-WEST has the endpoints.

Author: Thank you, @lihongan! I added call outs to the yaml file example, would you mind taking a look at those please?

The Ingress Operator uses `https://tagging.us-gov-west-1.amazonaws.com` as the tagging API endpoint if a tagging custom endpoint is configured that includes the string 'us-gov-east-1'.

[IMPORTANT]
====
Private, disconnected installs are not supported for AWS GovCloud when installing in the `us-gov-east-1` region.
====

.Example `route53` configuration
[source,yaml]
----
platform:
  aws:
    region: us-gov-west-1
    serviceEndpoints:
    - name: ec2
       url: https://ec2.us-gov-west-1.amazonaws.com
    - name: elasticloadbalancing
       url: https://elasticloadbalancing.us-gov-west-1.amazonaws.com
    - name: s3
       url: https://s3.us-east-1.amazonaws.com
    - name: tagging

>        url: https://tagging.us-east-1.amazonaws.com

Reviewer 2: the url should be `https://tagging.us-gov-west-1.amazonaws.com`

Author: Thank you, @lihongan! I updated that URL and changed the one for s3 to `https://route53.us-gov.amazonaws.com`.

// Module included in the following assemblies:
//
// * installing/installing_aws/installing-aws-government-region.adoc

[id="nw-endpoint-route53_{context}"]
= Ingress Operator endpoint configuration for AWS Route53

If you install in either Amazon Web Services (AWS) GovCloud (US) US-West or US-East region, the Ingress Operator uses `us-gov-west-1` region for Route53 and tagging API clients.

The Ingress Operator uses `https://tagging.us-gov-west-1.amazonaws.com` as the tagging API endpoint if a tagging custom endpoint is configured that includes the string 'us-gov-east-1'.

> For more information on AWS GovCloud (US) Endpoints, see the link:https://docs.aws.amazon.com/govcloud-us/latest/UserGuide/using-govcloud-endpoints.html[AWS GovCloud (US) Service Endpoints documentation].

Reviewer 3: ```suggestion
For more information on AWS GovCloud (US) Endpoints, see the link:https://docs.aws.amazon.com/govcloud-us/latest/UserGuide/using-govcloud-endpoints.html[Service Endpoints] in the AWS documentation about GovCloud (US).
```

[IMPORTANT]
====

> Private, disconnected installs are not supported for AWS GovCloud when installing in the `us-gov-east-1` region.

Reviewer 3: ```suggestion
Private, disconnected installations are not supported for AWS GovCloud when you install in the `us-gov-east-1` region.
```

====

.Example Route53 configuration
[source,yaml]
----
platform:
  aws:
    region: us-gov-west-1
    serviceEndpoints:
    - name: ec2
       url: https://ec2.us-gov-west-1.amazonaws.com
    - name: elasticloadbalancing
       url: https://elasticloadbalancing.us-gov-west-1.amazonaws.com

>     - name: s3

Reviewer 2: the name should be `- name: route53`, others look good. Thank you.

Author: Thank you! I'll update that now.

       url: https://route53.us-gov.amazonaws.com <1>
    - name: tagging
       url: https://tagging.us-gov-west-1.amazonaws.com <2>
----
<1> Route53 defaults to `https://route53.us-gov.amazonaws.com` for both AWS GovCloud (US) regions.

> <2> Only US-West has endpoints for tagging.

Reviewer 3: Maybe
```suggestion
<2> Only the US-West region has endpoints for tagging. Omit this parameter if your cluster is in another region.
```
